projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/jetpack-contact-form/ui-handler/with-ui-handler-data-provider.tsx

```diff
@@ -173,14 +175,50 @@ const withUiHandlerDataProvider = createHigherOrderComponent( BlockListBlock =>
 					// Update the list of current valid blocks
 					currentListOfValidBlocks.current = validBlocks;
 				}
+
+				// Final form adjustments (only when the request is done)
+				if ( isRequestDone ) {
+					/*
+					 * Inspect generated blocks list,
+					 * checking if the jetpack/button block:
+					 * - if it exists twice or more, remove the first one.
+					 * - if it does not exist, create one.
+					 */
+					const allButtonBlocks = validBlocks.filter( block => block.name === 'jetpack/button' );
+					currentListOfValidBlocks.current = currentListOfValidBlocks.current || [];
+					if ( allButtonBlocks.length > 1 ) {
+						// Remove all button blocks, less the last one.
+						currentListOfValidBlocks.current = currentListOfValidBlocks.current.filter(
+							( block, i ) => {
+								return block.name !== 'jetpack/button' || i === allButtonBlocks.length - 1;
```

Comment:
This line seems a bit stranger to me. 

Suppose we have 5 fields in the form and then 2 buttons at the end. The `i` value for them would be 5 and 6, right? Also, `allButtonBlocks` will have 2 elements, so `allButtonBlocks.length` will be 2. In that case, `i === allButtonBlocks.length - 1` will not be true form a button, but for a random field on the list.

Idea: what about filtering out all the buttons, and then appeding the last item of `allButtonBlocks` to the list?

Reply:
You're right. Let me update it

Reply:
Just to confirm, I tested it asking for 2 fields and 2 buttons, and I end up with no buttons, despite the buttons being on the completion response.

Reply:
https://github.com/Automattic/jetpack/pull/32581/commits/3609a0d3c21fb95fbc119d3b8ce6fedf8698e914


https://github.com/Automattic/jetpack/assets/77539/1cc6b6be-7a46-4411-88ad-b34a51daeb15